### PR TITLE
Improve partial parsing docs

### PR DESCRIPTION
**Pull request by tatiana**
Improves docs to highlight the limitation of the parsing parsing approach (introduced in #800), following up the feedback on #722

### CODE REVIEW

**Comment by tatiana** [comment]
@dwreeves if you have a chance, I'd love your feedback on this

File: docs/configuration/project-config.rst
Comment (dwreeves) [inline docs/configuration/project-config.rst:28]: The exception to this is that you can copy+paste the YAML generated by Cosmos, but I feel like that's splitting hairs and probably isn't worth mentioning.

**Comment by dwreeves** [comment]
One thing that is probably worth mentioning: Partial parsing requires the same `--vars`, `--target`, `--profile`, and env vars (as called by the `env_var()` macro) to be equal in each run, otherwise it doesn't work. [This specifically includes `profiles.yml` env vars](https://github.com/dbt-labs/dbt-core/blob/c6c0c79216a3c8032d5ea5ed608ca2ef7bdacc9a/core/dbt/parser/manifest.py#L838C1-L844C65).

For practical purposes, this means that the connection that generates the `manifest.json` must also be the same that actually utilizes it. So if you have for example a `CICD_USER` which runs `dbt parse` and an` AIRFLOW_USER` that actually executes in Cosmos, this won't work.

This also may end up meaning that users relying on `--vars` may end up running into issues (I have an idea for `1.5.0` to work around that a bit...).

**Review by pankajastro — APPROVED** [review]
LGTM

**Comment by tatiana** [comment]
@dwreeves, thanks for the detailed review; I tried to update with your feedback! Please let me know if I missed anything.
@pankajastro I made a few changes based on PR #800. Could you confirm that you're still happy with it?

**Comment by dwreeves** [comment]
Looks good but I've been out of the loop for the last month or two, and didn't realize you worked on a solution to make partial parsing better. 👀 Very nice!!! I need to catch up.

File: docs/configuration/partial-parsing.rst
Comment (tatiana) [reply docs/configuration/partial-parsing.rst]: Addressed in 081beccd1f25de45529972067d1e1a7d72d6fc73

File: docs/configuration/partial-parsing.rst
Comment (tatiana) [reply docs/configuration/partial-parsing.rst]: Thanks, @pankajkoti , addressed in 081beccd1f25de45529972067d1e1a7d72d6fc73

**Comment by tatiana** [comment]
Since this relies on the changes of #904, and once this PR is merged, it will be published into Cosmos official docs, I'll wait for #904 to be merged beforehand

**Pull request merged**
